jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1655,6 +1658,32 @@ void java_bytecode_parsert::rinner_classes_attribute(
   }
 }
 
+/// Corresponds to the element_value structure
+/// Described in Java 8 specification 4.7.5
+/// https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.7.5
+/// Parses the Exceptions attribute for the current method,
+/// which contains a table of the exceptions the method throws.
+void java_bytecode_parsert::rexceptions_attribute(methodt &method, const u4 &attribute_length)
+{
+  std::string name = method.name.c_str();
+  u2 number_of_exceptions = read_u2();
+  method.throws_exception_table.resize(number_of_exceptions);
```

Comment:
I think this is actually introducing problems which you have to deal with later.

If my understanding of how this `rexceptions_attribute` function is working is correct, then it is creating `number_of_exceptions` empty entries in the `throws_exception_table` and then adding the actual exceptions after this. Resulting in a vector which is twice as long as you expect, which contains junk empty items.

If you delete this line, then there will be no empty exceptions in the vector and you won't need to check for empty names when you get them out later.